app/services/foreman_discovery/host_converter.rb

```diff
@@ -52,14 +52,25 @@ def self.unused_ip_for_subnet(subnet, mac, existing_ip)
     end
   end
 
+  def self.ip_for_subnet(subnet, mac, ip) 
+    if ip
+      ipam = subnet.unused_ip(mac)
+
+      return ip if ipam.ip_include?(ip)
+    end
+
+    return unused_ip_for_subnet(subnet, mac, ip)
+  end
+
```

Comment:
Can you write this in a more Ruby fashion:


```ruby
  return subnet&.unused_ip(mac)&.ip_include?(ip) if ip

  unused_ip_for_subnet(subnet, mac, ip)
```

Reply:
```ruby
  return subnet&.unused_ip(mac)&.ip_include?(ip) if ip
```

Wouldn't this return `ip_include?(ip)` which is essentially a boolean?

Reply:
Sorry yeah what I meant was:

```ruby
  return ip if ip && subnet&.unused_ip(mac)&.ip_include?(ip)

  unused_ip_for_subnet(subnet, mac, ip)
```

Reply:
Sure, done